### DISCUSSION TITLE
Background playback stops randomly

### DISCUSSION THREAD

**Discussion by hachuwaka** [libre-tube/LibreTube#6017] [Q&A] [answered]
### Steps to reproduce
(pip mode disabled) 
1. Play a video
2. Go to the home screen
3. The video starts playing on the background as audio
4. The video stops playing after a few seconds, and only resumes either by chance, by hitting pause/play or by going into the app

### Expected behavior

Continued playback should happen, as it does on YouTube premium or on similar apps like NewPipe/Tubular

### Actual behavior

Background playback stops after a few seconds 

### LibreTube version

0.23.2

### Android version

Android 14 QPR2

### Other details

![Screenshot_20240509-075445_Kiwi_Browser](https://github.com/libre-tube/LibreTube/assets/169334402/7612d8e9-16f5-4b79-bb4f-c307ced63035)

As you can see, quick settings show that the video is, or at least should be, playing. 
Battery saver is off and the app has permissions to run on the background, in case you were wondering. Also the thing that it doesn't happen with NewPipe, also piped-based.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
Can't reproduce. Did you disable all kinds of battery optimizations? Dis you try using different instances?

**Reply by hachuwaka** [reply]
Yes to both things; I don't have any battery optimization on and I've just tried two other instances just to make sure

**Comment by hachuwaka** [comment]
https://github.com/libre-tube/LibreTube/assets/169334402/780af5fd-e1c1-4519-b31b-522723c2d763
There you go. It happens with every video, every instance, with background playback, with both the screen on and off. I know it may seem like a problem on my side, but I guess if that were true I'd be equally screwed with NewPipe too but NewPipe works. 

I'm not saying that other apps are better, more like the opposite. I like libretube so much I'd love to be able to use it as my go-to app for YouTube, but I can't because of this. 
Needless to say, thanks for your time and work!

**Reply by ryboto** [reply]
This is also happening for me. Background playback stops randomly and I can only restart playback at all by switching to video playback. It will not then switch back to audio only. This has been happening since around the time we had to switch to HLS. At that time when I tried Audio playback I would get a "Source Error" message, now it just stops with no erorr.

**Comment by Wurftaube** [comment]
Same issue here. Galaxy Note 9, Android 10, Libretube 023.2.

**Comment by Guuustav** [comment]
Same issue here. Galaxy S23 Ultra, Android 14, Libretube 023.2. Has anyone found a solution?

**Comment by aaasajsldfjsladkfj** [comment]
Same problem. The video in the background pauses after a while. Xiaomi 14, Android 15, LibreTube 0.26.0.

**Comment by Dzuchun** [comment]
Exact same situation with Pixel 7 and GrapheneOS (Android 15)

— LibreTube randomly stops playback while in background
— Once navigated to app, playback restarts immediately
— Other piped-based apps don't do that

What I tried:
— Toggling HLS
— Using all intances presented in the app
— Disabling battery optimizations
— Disabling Graphene's battery optimizer (global setting)
— Enabling Graphene's exploit protection compatibility mode
— Disabling traffic limitations
— Allowing app's execution in the background in warious ways
— Reinstalling the app, clearing cache and storage
— Installing via released .apk (i.e. directly, with no app registry in between)
— Disabling developer options
— Looking through log files at the time of playback stop (not much I can understand, unfortunately; can provide in case needed)
— Allowing/disallowing PIP (PIP is not my usecase, I'm just desperate at this point)

Just like the OP, I find LibreTube comfortable and sincerely wish to keep using it.
In my case however, I'm migrating to a new phone&system (from OnePlus 7 & Oxygen), which had no aforementioned issues, like, yesterday. So this bug might be hard to reproduce.

**Comment by Bnyro** [comment, marked as answer]
Duplicate of https://github.com/libre-tube/LibreTube/issues/5141 and resolved in https://github.com/libre-tube/LibreTube/pull/6732 (not released yet).
